arch/x86/kvm/pkvm/pgtable.c

```diff
@@ -752,3 +773,48 @@ int pkvm_pgtable_set_owner(struct pkvm_pgtable *pgt, unsigned long vaddr,
 
 	return pkvm_pgtable_walk(pgt, vaddr, size, &walker);
 }
+
+/**
+ * pkvm_pgtable_check_and_map() - Validates the owner id before maps a physical
+ *				  address range in a page table.
```

Comment:
pkvm_pgtable_check_and_map() does the same thing pkvm_pgtable_map() does, plus something extra. So it's better to keep its documentation in line with the documentation of pkvm_pgtable_map(), e.g.: "Install virtual addr to physical addr mappings in a pkvm pgtable, validating the page owner id before installing a mapping." ? And similarly for the rest of the documentation below (for instance, the descriptions of `@pgt`, `@vaddr`, `@phys`, `@size` args can be just copied from the pkvm_pgtable_map() documentation)?

Also, it's probably better to define pkvm_pgtable_check_and_map() after pkvm_pgtable_map(), not somewhere at the end of the file?